src/app/core/components/thread/thread.component.ts

```diff
@@ -61,29 +67,71 @@ export class ThreadComponent implements AfterViewInit, OnDestroy {
   readonly itemRoute$ = this.threadService.threadInfo$.pipe(
     map(t => (t ? rawItemRoute('activity', t.itemId) : undefined)),
   );
-
-  private subscription?: Subscription;
+  readonly isThreadStatusOpened$ = this.threadService.threadInfo$.pipe(
+    map(t => (t ? [ 'waiting_for_participant', 'waiting_for_trainer' ].includes(t.status) : false)),
+  );
+  readonly isCurrentUserThreadParticipant$ = this.userCache$.pipe(
+    map(userCache => userCache.some(u => u.isCurrentUser && u.isThreadParticipant)),
+  );
+  allowToOpenThreadState$ = merge(
+    of(readyState(false)),
+    combineLatest([
+      this.threadService.threadInfo$.pipe(
+        filter(isNotUndefined),
+        map(t => [ 'waiting_for_participant', 'waiting_for_trainer' ].includes(t.status)),
+        filter(isThreadOpened => !isThreadOpened),
+      ),
+      this.discussionService.visible$.pipe(distinctUntilChanged(), filter(visible => visible)),
+    ]).pipe(
+      withLatestFrom(
+        this.itemRoute$.pipe(filter(isNotUndefined)),
+        this.isCurrentUserThreadParticipant$,
+        this.groupWatchingService.watchedGroup$,
+      ),
+      switchMap(([ , itemRoute, isCurrentUserThreadParticipant, watchedGroup ]) =>
+        this.getItemByIdService.get(itemRoute.id, watchedGroup?.route.id).pipe(
+          mapToFetchState(),
+          mapStateData(itemData =>
+            isCurrentUserThreadParticipant && itemData.permissions.canRequestHelp
+            || !!watchedGroup && allowsWatchingAnswers(itemData.permissions)
+          ),
+        ),
+      ),
+    ),
+  );
 
   constructor(
     private threadService: ThreadService,
     private discussionService: DiscussionService,
     private userSessionService: UserSessionService,
     private groupWatchingService: GroupWatchingService,
     private userService: GetUserService,
+    private getItemByIdService: GetItemByIdService,
+    private actionFeedbackService: ActionFeedbackService,
     private fb: FormBuilder,
   ) {}
 
   ngAfterViewInit(): void {
-    this.subscription = this.state$.pipe(
+    this.state$.pipe(
       readyData(),
       withLatestFrom(this.discussionService.visible$),
       filter(([ events, visible ]) => visible && events.length > 0),
       delay(0),
+      takeUntil(this.destroyed$),
     ).subscribe(() => this.scrollDown());
+
+    this.isThreadStatusOpened$.pipe(delay(0), takeUntil(this.destroyed$)).subscribe(isThreadStatusOpened => {
```

Comment:
Subscriptions should be registered so that it can be unsubscribed on destroy.

Reply:
But it destroyed by `this.destroyed$`

Reply:
Yes sure, sorry I missed the takeUntil. But I have to recheck because I think we tried to take the approach always unsubscribe for other subscriptions... it is just a matter a consistency, I need to check.